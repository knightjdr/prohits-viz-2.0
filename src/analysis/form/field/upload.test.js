import React from 'react';
import { mount } from 'enzyme';

import Upload from './upload';
import TestForm from './__mocks__/form-wrapper';

const inputChange = jest.fn();
const onChange = jest.fn();
const onSubmit = jest.fn();

const file = new File([''], 'samplefile.txt', { type: 'text/plain' });
file.uid = 'rc-upload-sampleFile';

describe('Upload', () => {
  describe('without error', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mount(
        <TestForm
          input={{
            change: inputChange,
            value: undefined,
          }}
          meta={{ error: '', touched: false, warning: '' }}
          onSubmit={onSubmit}
        >
          <Upload
            input={{}}
            meta={{}}
            name="TestFile"
            onChange={onChange}
            required
            style={{ backgroundColor: '#000' }}
          />
        </TestForm>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should set initial file list', () => {
      const upload = wrapper.find('Upload').first();
      expect(upload.props().fileList).toEqual([]);
    });

    it('should add custom style', () => {
      const uploadStyle = wrapper.find('Button').props().style;
      expect(uploadStyle).toHaveProperty('backgroundColor', '#000');
    });

    it('should call change when file added', () => {
      onChange.mockClear();
      const input = wrapper.find('input');
      input.simulate('change', { target: { files: [file] } });
      expect(onChange).toHaveBeenCalled();
    });

    it('should set input value via input.value prop', () => {
      wrapper.setProps({
        input: {
          change: inputChange,
          value: [file],
        },
      });
      const upload = wrapper.find('Upload').first();
      expect(upload.props().fileList).toEqual([file]);
    });

    it('should call submit on button click', () => {
      onSubmit.mockClear();
      const button = wrapper.find('button').at(1);
      button.simulate('submit');
      expect(onSubmit).toHaveBeenCalledTimes(1);
    });
  });

  describe('submit error', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = mount(
        <TestForm
          input={{
            change: inputChange,
            value: [],
          }}
          meta={{ error: 'Error message', touched: true, warning: '' }}
        >
          <Upload
            input={{}}
            meta={{}}
            name="TestFile"
            onChange={onChange}
            required
            style={{}}
          />
        </TestForm>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have error message', () => {
      const formItem = wrapper.find('FormItem');
      expect(formItem.props().help).toBe('Error message');
    });

    it('should set validation status', () => {
      const formItem = wrapper.find('FormItem');
      expect(formItem.props().validateStatus).toBe('error');
    });
  });
});
