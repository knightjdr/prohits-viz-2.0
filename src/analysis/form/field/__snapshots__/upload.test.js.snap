// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select submit error should match snapshot 1`] = `
<TestForm
  input={
    Object {
      "change": [Function],
      "value": Array [],
    }
  }
  meta={
    Object {
      "error": "Error message",
      "touched": true,
      "warning": "",
    }
  }
  onSubmit={[Function]}
>
  <Form
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
    prefixCls="ant-form"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <CustomUpload
        input={
          Object {
            "change": [Function],
            "value": Array [],
          }
        }
        label={null}
        meta={
          Object {
            "error": "Error message",
            "touched": true,
            "warning": "",
          }
        }
        name="TestFile"
        onChange={[Function]}
        required={true}
        style={Object {}}
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          help="Error message"
          label={null}
          prefixCls="ant-form"
          validateStatus="error"
        >
          <Row
            className="ant-form-item ant-form-item-with-help"
            gutter={0}
          >
            <div
              className="ant-row ant-form-item ant-form-item-with-help"
            >
              <Col
                className="ant-form-item-control-wrapper"
                key=".$wrapper"
              >
                <div
                  className="ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control has-error"
                  >
                    <span
                      className="ant-form-item-children"
                    >
                      <Upload
                        accept=""
                        action=""
                        beforeUpload={[Function]}
                        className=""
                        data={Object {}}
                        disabled={false}
                        fileList={Array []}
                        listType="text"
                        multiple={true}
                        name="file"
                        onChange={[Function]}
                        prefixCls="ant-upload"
                        showUploadList={true}
                        supportServerRender={true}
                        type="select"
                      >
                        <span
                          className=""
                        >
                          <div
                            className="ant-upload ant-upload-select ant-upload-select-text"
                            style={
                              Object {
                                "display": "",
                              }
                            }
                          >
                            <Upload
                              accept=""
                              action=""
                              beforeUpload={[Function]}
                              component="span"
                              customRequest={null}
                              data={Object {}}
                              disabled={false}
                              fileList={Array []}
                              headers={Object {}}
                              listType="text"
                              multipart={false}
                              multiple={true}
                              name="file"
                              onChange={[Function]}
                              onError={[Function]}
                              onProgress={[Function]}
                              onReady={[Function]}
                              onStart={[Function]}
                              onSuccess={[Function]}
                              prefixCls="ant-upload"
                              showUploadList={true}
                              supportServerRender={true}
                              type="select"
                              withCredentials={false}
                            >
                              <AjaxUploader
                                accept=""
                                action=""
                                beforeUpload={[Function]}
                                component="span"
                                customRequest={null}
                                data={Object {}}
                                disabled={false}
                                fileList={Array []}
                                headers={Object {}}
                                listType="text"
                                multipart={false}
                                multiple={true}
                                name="file"
                                onChange={[Function]}
                                onError={[Function]}
                                onProgress={[Function]}
                                onReady={[Function]}
                                onStart={[Function]}
                                onSuccess={[Function]}
                                prefixCls="ant-upload"
                                showUploadList={true}
                                supportServerRender={true}
                                type="select"
                                withCredentials={false}
                              >
                                <span
                                  className="ant-upload"
                                  onClick={[Function]}
                                  onDragOver={[Function]}
                                  onDrop={[Function]}
                                  onKeyDown={[Function]}
                                  role="button"
                                  tabIndex="0"
                                >
                                  <input
                                    accept=""
                                    key="rc-upload-1547835505668-4"
                                    multiple={true}
                                    onChange={[Function]}
                                    style={
                                      Object {
                                        "display": "none",
                                      }
                                    }
                                    type="file"
                                  />
                                  <Button
                                    ghost={false}
                                    loading={false}
                                    prefixCls="ant-btn"
                                    style={Object {}}
                                  >
                                    <button
                                      className="ant-btn"
                                      onClick={[Function]}
                                      style={Object {}}
                                      type="button"
                                    >
                                      <FontAwesomeIcon$1
                                        border={false}
                                        className="FileInput-fa-icon"
                                        fixedWidth={false}
                                        flip={null}
                                        icon={
                                          Object {
                                            "icon": Array [
                                              384,
                                              512,
                                              Array [],
                                              "f319",
                                              "M384 121.9v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9zM248 160h136v328c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V24C0 10.7 10.7 0 24 0h200v136c0 13.2 10.8 24 24 24zm48 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z",
                                            ],
                                            "iconName": "file-plus",
                                            "prefix": "fas",
                                          }
                                        }
                                        key=".0"
                                        listItem={false}
                                        mask={null}
                                        name=""
                                        pull={null}
                                        pulse={false}
                                        rotation={null}
                                        size={null}
                                        spin={false}
                                        symbol={false}
                                        transform={null}
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className="svg-inline--fa fa-file-plus fa-w-12 FileInput-fa-icon"
                                          data-icon="file-plus"
                                          data-prefix="fas"
                                          role="img"
                                          style={Object {}}
                                          viewBox="0 0 384 512"
                                          xmlns="http://www.w3.org/2000/svg"
                                        >
                                          <path
                                            d="M384 121.9v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9zM248 160h136v328c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V24C0 10.7 10.7 0 24 0h200v136c0 13.2 10.8 24 24 24zm48 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z"
                                            fill="currentColor"
                                            style={Object {}}
                                          />
                                        </svg>
                                      </FontAwesomeIcon$1>
                                      <span
                                        key=".1"
                                      >
                                         Select File
                                      </span>
                                    </button>
                                  </Button>
                                </span>
                              </AjaxUploader>
                            </Upload>
                          </div>
                          <LocaleReceiver
                            componentName="Upload"
                            defaultLocale={
                              Object {
                                "previewFile": "Preview file",
                                "removeFile": "Remove file",
                                "uploadError": "Upload error",
                                "uploading": "Uploading...",
                              }
                            }
                          >
                            <UploadList
                              items={Array []}
                              listType="text"
                              locale={
                                Object {
                                  "previewFile": "Preview file",
                                  "removeFile": "Remove file",
                                  "uploadError": "Upload error",
                                  "uploading": "Uploading...",
                                }
                              }
                              onRemove={[Function]}
                              prefixCls="ant-upload"
                              progressAttr={
                                Object {
                                  "showInfo": false,
                                  "strokeWidth": 2,
                                }
                              }
                              showPreviewIcon={true}
                              showRemoveIcon={true}
                            >
                              <Animate
                                animation={Object {}}
                                className="ant-upload-list ant-upload-list-text"
                                component="div"
                                componentProps={Object {}}
                                onAppear={[Function]}
                                onEnd={[Function]}
                                onEnter={[Function]}
                                onLeave={[Function]}
                                transitionAppear={false}
                                transitionEnter={true}
                                transitionLeave={true}
                                transitionName="ant-upload-animate"
                              >
                                <div
                                  className="ant-upload-list ant-upload-list-text"
                                />
                              </Animate>
                            </UploadList>
                          </LocaleReceiver>
                        </span>
                      </Upload>
                    </span>
                    <Animate
                      animation={Object {}}
                      component=""
                      componentProps={Object {}}
                      key="help"
                      onAppear={[Function]}
                      onEnd={[Function]}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="show-help"
                    >
                      <AnimateChild
                        animation={Object {}}
                        key="help"
                        transitionAppear={true}
                        transitionEnter={true}
                        transitionLeave={true}
                        transitionName="show-help"
                      >
                        <div
                          className="ant-form-explain"
                          key="help"
                        >
                          Error message
                        </div>
                      </AnimateChild>
                    </Animate>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </FormItem>
      </CustomUpload>
      <button
        type="submit"
      />
    </form>
  </Form>
</TestForm>
`;

exports[`Select without error should match snapshot 1`] = `
<TestForm
  input={
    Object {
      "change": [Function],
      "value": undefined,
    }
  }
  meta={
    Object {
      "error": "",
      "touched": false,
      "warning": "",
    }
  }
  onSubmit={[Function]}
>
  <Form
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
    prefixCls="ant-form"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <CustomUpload
        input={
          Object {
            "change": [Function],
            "value": undefined,
          }
        }
        label={null}
        meta={
          Object {
            "error": "",
            "touched": false,
            "warning": "",
          }
        }
        name="TestFile"
        onChange={[Function]}
        required={true}
        style={
          Object {
            "backgroundColor": "#000",
          }
        }
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          help=""
          label={null}
          prefixCls="ant-form"
          validateStatus=""
        >
          <Row
            className="ant-form-item"
            gutter={0}
          >
            <div
              className="ant-row ant-form-item"
            >
              <Col
                className="ant-form-item-control-wrapper"
                key=".$wrapper"
              >
                <div
                  className="ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    >
                      <Upload
                        accept=""
                        action=""
                        beforeUpload={[Function]}
                        className=""
                        data={Object {}}
                        disabled={false}
                        fileList={Array []}
                        listType="text"
                        multiple={true}
                        name="file"
                        onChange={[Function]}
                        prefixCls="ant-upload"
                        showUploadList={true}
                        supportServerRender={true}
                        type="select"
                      >
                        <span
                          className=""
                        >
                          <div
                            className="ant-upload ant-upload-select ant-upload-select-text"
                            style={
                              Object {
                                "display": "",
                              }
                            }
                          >
                            <Upload
                              accept=""
                              action=""
                              beforeUpload={[Function]}
                              component="span"
                              customRequest={null}
                              data={Object {}}
                              disabled={false}
                              fileList={Array []}
                              headers={Object {}}
                              listType="text"
                              multipart={false}
                              multiple={true}
                              name="file"
                              onChange={[Function]}
                              onError={[Function]}
                              onProgress={[Function]}
                              onReady={[Function]}
                              onStart={[Function]}
                              onSuccess={[Function]}
                              prefixCls="ant-upload"
                              showUploadList={true}
                              supportServerRender={true}
                              type="select"
                              withCredentials={false}
                            >
                              <AjaxUploader
                                accept=""
                                action=""
                                beforeUpload={[Function]}
                                component="span"
                                customRequest={null}
                                data={Object {}}
                                disabled={false}
                                fileList={Array []}
                                headers={Object {}}
                                listType="text"
                                multipart={false}
                                multiple={true}
                                name="file"
                                onChange={[Function]}
                                onError={[Function]}
                                onProgress={[Function]}
                                onReady={[Function]}
                                onStart={[Function]}
                                onSuccess={[Function]}
                                prefixCls="ant-upload"
                                showUploadList={true}
                                supportServerRender={true}
                                type="select"
                                withCredentials={false}
                              >
                                <span
                                  className="ant-upload"
                                  onClick={[Function]}
                                  onDragOver={[Function]}
                                  onDrop={[Function]}
                                  onKeyDown={[Function]}
                                  role="button"
                                  tabIndex="0"
                                >
                                  <input
                                    accept=""
                                    key="rc-upload-1547835505668-1"
                                    multiple={true}
                                    onChange={[Function]}
                                    style={
                                      Object {
                                        "display": "none",
                                      }
                                    }
                                    type="file"
                                  />
                                  <Button
                                    ghost={false}
                                    loading={false}
                                    prefixCls="ant-btn"
                                    style={
                                      Object {
                                        "backgroundColor": "#000",
                                      }
                                    }
                                  >
                                    <button
                                      className="ant-btn"
                                      onClick={[Function]}
                                      style={
                                        Object {
                                          "backgroundColor": "#000",
                                        }
                                      }
                                      type="button"
                                    >
                                      <FontAwesomeIcon$1
                                        border={false}
                                        className="FileInput-fa-icon"
                                        fixedWidth={false}
                                        flip={null}
                                        icon={
                                          Object {
                                            "icon": Array [
                                              384,
                                              512,
                                              Array [],
                                              "f319",
                                              "M384 121.9v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9zM248 160h136v328c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V24C0 10.7 10.7 0 24 0h200v136c0 13.2 10.8 24 24 24zm48 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z",
                                            ],
                                            "iconName": "file-plus",
                                            "prefix": "fas",
                                          }
                                        }
                                        key=".0"
                                        listItem={false}
                                        mask={null}
                                        name=""
                                        pull={null}
                                        pulse={false}
                                        rotation={null}
                                        size={null}
                                        spin={false}
                                        symbol={false}
                                        transform={null}
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className="svg-inline--fa fa-file-plus fa-w-12 FileInput-fa-icon"
                                          data-icon="file-plus"
                                          data-prefix="fas"
                                          role="img"
                                          style={Object {}}
                                          viewBox="0 0 384 512"
                                          xmlns="http://www.w3.org/2000/svg"
                                        >
                                          <path
                                            d="M384 121.9v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9zM248 160h136v328c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V24C0 10.7 10.7 0 24 0h200v136c0 13.2 10.8 24 24 24zm48 140c0-6.6-5.4-12-12-12h-60v-60c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v60h-60c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h60v60c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12v-60h60c6.6 0 12-5.4 12-12v-40z"
                                            fill="currentColor"
                                            style={Object {}}
                                          />
                                        </svg>
                                      </FontAwesomeIcon$1>
                                      <span
                                        key=".1"
                                      >
                                         Select File
                                      </span>
                                    </button>
                                  </Button>
                                </span>
                              </AjaxUploader>
                            </Upload>
                          </div>
                          <LocaleReceiver
                            componentName="Upload"
                            defaultLocale={
                              Object {
                                "previewFile": "Preview file",
                                "removeFile": "Remove file",
                                "uploadError": "Upload error",
                                "uploading": "Uploading...",
                              }
                            }
                          >
                            <UploadList
                              items={Array []}
                              listType="text"
                              locale={
                                Object {
                                  "previewFile": "Preview file",
                                  "removeFile": "Remove file",
                                  "uploadError": "Upload error",
                                  "uploading": "Uploading...",
                                }
                              }
                              onRemove={[Function]}
                              prefixCls="ant-upload"
                              progressAttr={
                                Object {
                                  "showInfo": false,
                                  "strokeWidth": 2,
                                }
                              }
                              showPreviewIcon={true}
                              showRemoveIcon={true}
                            >
                              <Animate
                                animation={Object {}}
                                className="ant-upload-list ant-upload-list-text"
                                component="div"
                                componentProps={Object {}}
                                onAppear={[Function]}
                                onEnd={[Function]}
                                onEnter={[Function]}
                                onLeave={[Function]}
                                transitionAppear={false}
                                transitionEnter={true}
                                transitionLeave={true}
                                transitionName="ant-upload-animate"
                              >
                                <div
                                  className="ant-upload-list ant-upload-list-text"
                                />
                              </Animate>
                            </UploadList>
                          </LocaleReceiver>
                        </span>
                      </Upload>
                    </span>
                    <Animate
                      animation={Object {}}
                      component=""
                      componentProps={Object {}}
                      key="help"
                      onAppear={[Function]}
                      onEnd={[Function]}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="show-help"
                    />
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </FormItem>
      </CustomUpload>
      <button
        type="submit"
      />
    </form>
  </Form>
</TestForm>
`;
